### PR TITLE
Run parquet2hive after Main Summary

### PR DESCRIPTION
**Pull request by robotblake**
This adds a generic parquet2hive job and a call to it for the main_summary dataset.

### CODE REVIEW

**Comment by whd** [comment]
I think we should be setting your p2h job to be the upstream for the jobs that have main_summary as their current upstream (assuming they use the centralized hive metastore to access main_summary). I'm guessing we already have the problem with main summary that [we had with longitudinal](https://bugzilla.mozilla.org/show_bug.cgi?id=1355153). This dag structure appears to make explicit that problem, and if the table is unavailable while the msck repair is running, running it in parallel with existing downstream jobs could cause more failures. In the fix for longitudinal we just tacked the p2h call at the end of the longitudinal run script (https://github.com/mozilla/telemetry-airflow/pull/102), with the understanding that eventually we will use a proper p2h operator as described [here]( https://bugzilla.mozilla.org/show_bug.cgi?id=1356703).

**Comment by fbertsch** [comment]
I don't know if we should really be setting p2h upstream from the other jobs. They are all reading from the actual s3 files right now, and adding an extra 5+ hours delay to when they run may be detrimental. This should *eventually* be done, but not until we make metastore indexing much faster.

**Comment by whd** [comment]
@fbertsch his is why I predicated my comment with "assuming they use the centralized hive metastore to access main_summary", and did not explicitly r- this. @robotblake explained to me that downstream jobs do not use hive, and was going to comment to that effect, but it looks like you beat him to it.

File: dags/main_summary.py [hunk 1]
```diff
@@ -94,10 +94,22 @@
                       uri="https://raw.githubusercontent.com/mozilla/python_mozetl/master/mozetl/testpilot/txp_mau_dau.py",
                       dag=dag)
 
+t9 = EMRSparkOperator(task_id="p2h_main_summary",
+                      job_name="P2H for Main Summary",
+                      execution_timeout=timedelta(hours=4),
```
Comment (whd) [inline dags/main_summary.py]: Will this timeout be sufficient? @fbertsch has just mentioned 5+ hours as a possibility.

**Comment by fbertsch** [comment]
@whd I misunderstood (or misread). Thanks for clarifying.

Another point: We are going to be doing a big backfill next week (6 months of data). We don't want to call p2h after every single run. How should we handle this? One option is only run p2h if the $batchdate is $YESTERDAY. All historical runs will not reindex the metastore.

File: dags/main_summary.py (hunk 1 above)
Comment (robotblake) [inline dags/main_summary.py]: Ugh, yeah, should probably put it longer.

**Comment by robotblake** [comment]
@fbertsch Good point, I'll take a look at adding that.

**Comment by mreid-moz** [comment]
> We don't want to call p2h after every single run. How should we handle this?

Is it possible to check if hive already contains a given partition?

**Comment by fbertsch** [comment]
@robotblake, can we close this or fix it?

**Pull request closed without merging**